packages/appsync-modelgen-plugin/src/visitors/appsync-typescript-visitor.ts

```diff
@@ -117,7 +125,7 @@ export class AppSyncModelTypeScriptVisitor<
           },
           {
             name: 'mutator',
-            type: `(draft: MutableModel<${modelName}>) => MutableModel<${modelName}> | void`,
+            type: `(draft: MutableModel<${modelName}${readOnlyTypeLabels}>) => MutableModel<${modelName}${readOnlyTypeLabels}> | void`,
```

Comment:
These changes would need to be isolated under a feature flag.  (I think Aaron pointed out feature flag as well)

Reply:
@frimfram I've added a check for the feature flag, please verify. Thank you!